docs/specifications/tests/Methods.md

```diff
@@ -1,69 +1,284 @@
-   step 1.
+* The name of the child zone ("child zone").
+* The fact if the child zone exists from [BASIC01].
+* The test type, "undelegate test" or "normal test".
+* If *normal test*, the name servers to the parent zone as found 
+  in [BASIC01] ("parent NS").
+* If *undelegated test*, the submitted delegation data, name server names
+  and IP addresses for *child zone* ("undelegated data").
 
-### Method 4: Obtain "glue address records" from parent
+### Ordered description of steps to be taken to execute the method
 
-This method tries to obtain any glue address records from the delegation
-in the parent zone.
+1. *Normal test*. For *undelegated test* go to 2.
+   1. If child zone does not exists then return ERROR.
+   2. Create an SOA query for the *child zone* and send that one server of
+      *parent NS* with RD flag unset.
+   3. If the response response contains a referral to the child zone:
+      1. Extract the name server names from the RDATA of the NS records in
+         the authority section.
+      2. Repeat the SOA query for *child zone* to the other name servers in
+         "parent NS":
+	 1. If the response from a server has the AA bit set, ignore that
+	    response.
+	 2. If the response is a referral with other NS than found above, 
+            then add the name
+	    server names of the NS to the list of name servers of the
+	    delegation for the *child zone*.
+      3. Return the set of name servers (name server names).
+      4. Processing the steps is stopped.
+   4. If the response is authoritative (AA bit set) and the answer section
+      contains the SOA record of the child zone:
+      1. Repeat the query with the next server of *parent NS* until a 
+         server has responded with a referral or no more servers are available.
+      2. If a referral was found, go to step 3.
+      3. If no referral was found, send a query for the NS records instead,
+         and use them as if it was a referral.
+      4. Return the set of name servers (name server names).
+      5. Processing the steps is stopped.
 
-1. Query the servers in Method 2 for A and AAAA addresses.
-2. Record the unique IP addresses from the answers (both A and AAAA) in
-   the additional section, which are the glue address records for the
-   domain.
+2. *Undelegated test*.
 
-### Method 5: Obtain the name server address records from child
+   1. Collect the name server names from the *undelegated data*.
+   2. Return the set of name servers (name server names).
+   3. Processing the steps is stopped.
 
-This method tries to obtain the IP addresses for the name servers used in
-the child zone.
 
-1. Send an A query to all name servers obtained in Method 3.
-2. Record the list of unique IPv4 addreses in the answer section.
-3. Send an AAAA query to all name servers obtained in Method 3.
-4. Record the list of unique IPv6 addresses in the answer section.
+### Outcome(s)
```

Comment:
Come to think of it, the "outcome" of a method feels a bit weird. Return value would make more sense.